### ISSUE TITLE
Vertex property with a key containing a dot fails silently

### ISSUE THREAD

**Issue by pboos** [orientechnologies/orientdb#5838]
orientdb version: 2.2.0-beta

`setProperty("some.key", "value")` just fails silently. It does not set the value on the vertex. When doing `getProperty("some.key")` it returns null.

Example test:

```
    @Test
    public void testProperties() {
        String propertyKey = "property.test"; // removing the "." makes the test pass
        String propertyValue = "test";

        OrientVertex vertex = orientGraph.addVertex("V");
        vertex.setProperty(propertyKey, propertyValue);

        assertEquals(propertyValue, vertex.getProperty(propertyKey)); // --> here it returns null
    }
```

Workaround: replace . with -.

In `OSchemaShared.checkFieldNameIfValid()` it says checks for the following invalid characters:

```
if (c == ':' || c == ',' || c == ';' || c == ' ' || c == '%' || c == '=')
```

Maybe `.` is used internally? I didn't check further. But probably would be nice to have this fixed or at least throw so that one knows that property keys with a `.` are not allowed.

**Comment by nevers** [comment]
Had the same problem today. I replaced the dots with underscores, but it's a rather ugly hack.

**Comment by luigidellaquila** [comment]
This is fixed in v 3.0

Closing, thanks!

Luigi
